src/Raven.Server/Documents/Patch/JavaScriptUtils.cs

```diff
@@ -261,6 +263,20 @@ internal JsValue GetDocumentId(JsValue self, JsValue[] args)
 
         internal JsValue TranslateToJs(Engine engine, JsonOperationContext context, object o)
         {
+            if (o is TimeSeriesRetriever.TimeSeriesStreamingRetrieverResult tsrr)
+            {
+				// we are passing a streaming value to the JS engine, so we need
+				// to materalize all the results
+                
+                
+                var results = new DynamicJsonArray(tsrr.Stream);
+                var djv = new DynamicJsonValue
+                {
+                    ["Count"] = results.Count,
```

Comment:
1. Why do we have `Count` here? I see that in tests you do `ts.Results.length;`
2. Can we have some class that would represent the result, so we could use `nameof()` instead of hard coded strings?
3. Why the returned result is an object instead of an array? I see your comment above but I don't follow it - https://github.com/ravendb/ravendb/pull/12948/files#r744106385. Do we want to return an object to have the option to expand it in the future or something like that?

Reply:
1) To match the structure of the value in C#, where we have:
https://github.com/ravendb/ravendb/blob/0fa8dabaf777caf6dcd725b182d9eb4d41b13017/src/Raven.Client/Documents/Queries/TimeSeries/TimeSeriesAggregation.cs#L18

2) We can use `TimeSeriesAggregationResult`, I guess. 

3) Because without streaming, we are passing `TimeSeriesAggregationResult` to the JavaScript function, and we want to match that structure.

Reply:
Thanks for clarification. That makes sense. I pushed the fixes - https://github.com/ravendb/ravendb/pull/12948/commits/2e4da514539106587910301d67754035f4107582

One more question. Shouldn't we add `@projection: true`  to the metadata result? I see we do it in `TimeSeriesRetriever.MaterializeResults()` if we have a projection:

https://github.com/ravendb/ravendb/blob/21648565af49a7bbcc9c6adf2e9765223b2b2b36/src/Raven.Server/Documents/Queries/Results/TimeSeries/TimeSeriesRetriever.cs#L894-L900

Reply:
That isn't the end result, this is being passed to the JS engine, so the output of _that_ will add the projection

Reply:
Okay, I see.